net/MapViewController.m

```diff
@@ -126,6 +129,23 @@ -(void)viewWillAppear:(BOOL)animated
     [runner addTimer:self.timerForMapRenew forMode: NSDefaultRunLoopMode];
 }
 
+-(void)selectCurrentMarker
+{
+        if(self.isMarkerSelected == YES) {
```

Comment:
It can't be this way because self.mapView.selectedMarker automatically sets to "nil" after changing of view controller. So I must manually handle this behaviour by creating my own property.